### DISCUSSION TITLE
auth.role() deprecated for all uses or just RLS policies?

### DISCUSSION THREAD

**Discussion by milesingrams** [supabase/supabase#7291] [Questions] [answered]
I saw on the supabase docs that auth.role() was deprecated for RLS policies (https://supabase.com/docs/guides/auth/row-level-security#authrole)

I was going to use it in the body of a RPC function to do different things based on whether a user is anon or authenticated. Can I still use it for that purpose without worrying about it being removed? I feel like there are still many uses for getting the user auth role for granular control of what a function does.

**Comment by GaryAustin1** [comment, marked as answer]
The function is a few lines of code based on PostgREST processing the jwt, so if the function were to go away it is easily replaced by your own.   But a good question as it should have been made clear.

**Reply by milesingrams** [reply]
Thanks for the quick response! I now feel confident using knowing I can easily replace the function if it were removed. I still feel like it should remain. But I understand the strong recommendation not to use it in RLS because the alternative is much more efficient.
